src/utils/variables.ts

```diff
@@ -1,20 +1,10 @@
-import {Variables} from '@models/variable';
+import {VariableInForm, Variables} from '@models/variable';
 
-export const formatVariables = (list: any[]) => {
-  const variables: {
-    [key in string]: {
-      name: string;
-      value?: string;
-      type: string;
-      secretRef?: {
-        name: string;
-        key: string;
-      };
-    };
-  } = {};
+export const formatVariables = (list: VariableInForm[]) => {
+  const variables: Variables = {};
 
   list.forEach(item => {
-    if (item.secretRef || item.type === 'secretRef') {
+    if (item.secretRefName && item.secretRefKey && item.type === 'secretRef') {
```

Comment:
To confirm - did you check that's expected?

Reply:
Tested locally, should be good

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/25991946/234541299-948ad091-b278-4398-ac63-160813048cf0.png">